### PR TITLE
Add new test for testing the home node feature in linux kernel

### PR DESCRIPTION
**Pull request by Geet2**
mm/mempolicy: add set_mempolicy_home_node syscall
This syscall can be used to set a home node for the MPOL_BIND and MPOL_PREFERRED_MANY memory policy. The syscall allows specifying a home node/preferred node from which kernel will fulfill memory allocation requests first.

Link: https://lkml.kernel.org/r/20211202123810.267175-3-aneesh.kumar@linux.ibm.com

### CODE REVIEW

**Comment by Geet2** [comment]
Output :
```

JOB ID     : 2162d1eb890ec19dd8dd8b2bd565406b77dd9301
JOB LOG    : /home/geet/avocado-fvt-wrapper/results/job-2023-04-19T05.29-2162d1e/job.log
Task exception was never retrieved
future: <Task finished coro=<StatusServer.serve_forever() done, defined at /usr/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/status/server.py:40> exception=AttributeError("'Server' object has no attribute 'serve_forever'",)>
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/avocado_framework-100.1-py3.6.egg/avocado/core/status/server.py", line 44, in serve_forever
    await self._server_task.serve_forever()
AttributeError: 'Server' object has no attribute 'serve_forever'
 (1/2) homenode.py:HomeNodeTest.test;run-page_type-normal_page-5c40: STARTED
 (1/2) homenode.py:HomeNodeTest.test;run-page_type-normal_page-5c40:  PASS (0.08 s)
 (2/2) homenode.py:HomeNodeTest.test;run-page_type-hugepage-a709: STARTED
 (2/2) homenode.py:HomeNodeTest.test;run-page_type-hugepage-a709:  PASS (0.08 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/geet/avocado-fvt-wrapper/results/job-2023-04-19T05.29-2162d1e/results.html
JOB TIME   : 18.33 s
```

File: memory/homenode.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    def copyutil(self, file_name):
+        shutil.copyfile(self.get_data(file_name),
+                        os.path.join(self.teststmpdir, file_name))
+
+    def setUp(self):
+        smm = SoftwareManager()
+        self.dist = distro.detect()
+
+        self.nr_pages = self.params.get('nr_pages', default=100)
+        self.map_type = self.params.get('map_type', default='private')
```
Comment (sacsant) [inline memory/homenode.py:40]: Can you update the YAML file to also cover shared option as well? I see the testcase already has a code to cover shared option. Rest looks good.

File: memory/homenode.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            pkgs.extend(['libnuma-devel'])
+            if self.dist.version >= 15:
+                pkgs.extend(['libhugetlbfs-devel'])
+            else:
+                pkgs.extend(['libhugetlbfs-libhugetlb-devel'])
+
+        hp_check = 0
+        if self.hpage:
+            hp_size = memory.get_huge_page_size()
+            hp_check += int(genio.read_file(
```
Comment (PraveenPenguin) [inline memory/homenode.py:63]: seems variable in confusing can we have some meaningful variable name

File: memory/homenode.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                    "libhugetlbfs packages are unavailable RHEL 9.x onwards.")
+            pkgs.extend(['numactl-devel', 'libhugetlbfs-devel'])
+        elif self.dist.name == "SuSE":
+            pkgs.extend(['libnuma-devel'])
+            if self.dist.version >= 15:
+                pkgs.extend(['libhugetlbfs-devel'])
+            else:
+                pkgs.extend(['libhugetlbfs-libhugetlb-devel'])
+
+        hp_check = 0
```
Comment (PraveenPenguin) [inline memory/homenode.py:60]: this assignment is not require

File: memory/homenode.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            if hp_check < self.nr_pages:
+                self.cancel('Not enough pages to be configured on nodes')
+
+        for package in ['gcc', 'make']:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+        for file_name in ['homenode.c', 'Makefile']:
+            self.copyutil(file_name)
+
```
Comment (PraveenPenguin) [inline memory/homenode.py:75]: please remove this whitespace

File: memory/homenode.py.data/homenode.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+        nr_pages: 1024
+    hugepage:
+        h_page: True
+        nr_pages: 2
+maptype: !mux
+    shared:
+	maptype: shared
+    private:
+	maptype: private
+
```
Comment (PraveenPenguin) [inline memory/homenode.py.data/homenode.yaml:12]: please remove this whitespace

**Review by PraveenPenguin — COMMENTED** [review]
@Geet2 thanks for patch can you please address style comments

File: memory/homenode.py.data/homenode.yaml (hunk 5 above)
Comment (Geet2) [reply memory/homenode.py.data/homenode.yaml]: done

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**